### PR TITLE
update k8s min version to 1.26.x and go deps to 1.27.x

### LINKED ISSUE #2821: Bump our min k8s version to v1.26
As per the [release schedule](https://github.com/knative/community/blob/main/mechanics/RELEASE-SCHEDULE.md)

Prior changes https://github.com/knative/pkg/pull/2745

### PR DESCRIPTION
**Pull request by ReToCode**
# Changes
- Bumps the k8s min-version to 1.26.x
- While using the go deps of 1.27.x, according to https://github.com/knative/pkg/pull/2832#discussion_r1334418947

Fixes #2821

**Release Note**
```release-note
The minimal required kubernetes version to use `knative/pkg` is now v1.26.0`
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -40,11 +40,11 @@ require (
...
-	k8s.io/api v0.26.5
-	k8s.io/apiextensions-apiserver v0.26.5
-	k8s.io/apimachinery v0.26.5
-	k8s.io/client-go v0.26.5
-	k8s.io/code-generator v0.26.5
+	k8s.io/api v0.26.9
+	k8s.io/apiextensions-apiserver v0.26.9
+	k8s.io/apimachinery v0.26.9
+	k8s.io/client-go v0.26.9
+	k8s.io/code-generator v0.26.9
```
Comment (dprotaso) [inline go.mod:47]: We can bump to the v0.27.x since there's a compatibility with N & N-1

**Comment by ReToCode** [comment]
@dprotaso bumped go deps to 1.27.x. There are some "breaking" changes though, but not sure if those changes are going to be an issue somewhere:

https://github.com/knative/pkg/pull/2832/commits/6813aaab36f3bc24830d04b4150f862918fcbb65#diff-d32c3a9bdf258149fea763b4cb30d35ebafcf97e77b1c7694583e1425c661702L71

https://github.com/knative/pkg/pull/2832/commits/6813aaab36f3bc24830d04b4150f862918fcbb65#diff-b388631fadc29258560f1547149fdd6d83616ebeb559ac009798a9869bb5b3ddL99

And some client types are **now only** available as `v2alpha1` (`v1alpha1` was dropped):
https://github.com/knative/pkg/pull/2832/commits/6813aaab36f3bc24830d04b4150f862918fcbb65#diff-724fcf837c9df0942c8ffea6c96ae7f9ece5e68a39ad6e1a9d8c16f8db03eef0

**Comment by ReToCode** [comment]
Hm, any ideas @pierDipi, see downstream test failures and [comment above](https://github.com/knative/pkg/pull/2832#issuecomment-1742545988)

**Comment by dprotaso** [comment]
> There are some "breaking" changes though, but not sure if those changes are going to be an issue somewhere:

It's usually how it is

**Pull request merged**